File: app/src/main/java/joebruckner/lastpick/ActivityModule.java
package joebruckner.lastpick;

import android.content.Context;
import android.support.v7.app.AppCompatActivity;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;

@Module
public class ActivityModule {
    private AppCompatActivity activity;

    public ActivityModule(AppCompatActivity activity) {
        this.activity = activity;
    }

    @Provides
    @ActivityScope
    AppCompatActivity provideActivity() {
        return activity;
    }

    @Provides
    @ActivityScope
    @Named("Activity")
    Context provideContext() {
        return activity;
    }

}

File: app/src/main/java/joebruckner/lastpick/ActivityModule.kt
package joebruckner.lastpick

import android.content.Context
import android.support.v7.app.AppCompatActivity
import dagger.Module
import dagger.Provides
import javax.inject.Named

@Module
class ActivityModule(val activity: AppCompatActivity) {

    @Provides
    @ActivityScope
    fun provideActivity(): AppCompatActivity {
        return activity
    }

    @Provides
    @ActivityScope
    @Named("Activity")
    fun provideContext(): Context {
        return activity
    }

}